client/cli/src/config.rs

```diff
@@ -152,11 +153,18 @@ pub trait CliConfiguration: Sized {
 			.unwrap_or(Default::default()))
 	}
 
+	/// Get the database backend variant.
+	///
+	/// By default this is retrieved from `ImportParams` if it is available. Otherwise its `None`.
+	fn database(&self) -> Option<Database> {
+		self.import_params().map(|x| x.database())
+	}
```

Comment:
```suggestion
	fn database(&self) -> Result<Option<Database>> {
		Ok(self.import_params().map(|x| x.database()))
	}
```

All methods in this trait (except *_params) must be `Result<_>` so an application that wants to override can produce a CLI error while parsing if they want.